core/trino-main/src/main/java/io/trino/operator/SourceOperator.java

```diff
@@ -28,4 +28,12 @@
     Supplier<Optional<UpdatablePageSource>> addSplit(Split split);
 
     void noMoreSplits();
+
+    /**
+     * This method will be called to unblock any blocking operation that operator is doing.
+     * In order to unblock blocking operation this method is called in separate thread.
+     * Once this method is called, it is known that soon {@link close()} method will be called,
+     * so output is no longer needed.
+     */
+    default void cancel() {}
```

Comment:
```
This method is called before {@link #close()} being called. It is called asynchronously, so it can be called also
while the operator is performing other operations. [...]
```

Reply:
To further expand on @findepi's point. We would need to emphasize that this API will be called sometime in the vicinity of a close(), either before, during, or after.